docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+
+The best example of this is allowing the top-level Terraform Schema Model for a given Terraform Resource to map onto the `properties` object that exists in most payloads in Azure Resource Manager - allowing for fields which exist at the root of the Terraform Schema to be mapped to a nested object.
+
+Each ModelToModel Field Mapping must have an associated ModelToModel mapping, these are used together:
+
+* A ModelToModel Field Mapping is used to output the call to the mapping function itself.
+* A ModelToModel Mapping is used to output the mapping function, the fields which should be mapped are then inferred from the ModelToModel Field Mappings.
+
+> **Note:** A complete end-to-end example can be found below under `Example`.
+
+Model to Model Field Mappings support the following properties:
+
+* `schemaModelName` - specifies the name of the Terraform Schema Model that should be mapped onto the Field within the SDK Model.
+* `sdkModelName` - specifies the name of the SDK Model containing the SDK Field that the Terraform Schema Model should be mapped onto.
+* `sdkFieldPath` - specifies the Path to the Field within the SDK Model where the Terraform Schema Model should be mapped onto.
+
+An example of a Model to Model Field Mapping can be seen below: 
+
+```json
+{
+  "fieldMappings": [
+    {
+      "type": "ModelToModel",
+      "modelToModel": {
+        "schemaModelName": "SolarSystemResourceSchema",
+        "sdkModelName": "SolarSystem",
+        "sdkFieldName": "Properties"
+      }
+    }
+  ]
+}
+```
+
+This would allow the Terraform Generator to output:
+
+```go
+func (r VirtualMachineResource) mapVirtualMachineResourceSchemaToVirtualMachineProperties(input VirtualMachineResourceSchema, output *virtualmachines.VirtualMachine) error {
+    if output.Properties == nil {
+        output.Properties = &virtualmachines.VirtualMachineProperties{}
+    }
+    if err := r.mapVirtualMachineResourceSchemaToVirtualMachineProperties(input, output.Properties); err != nil {
+        return fmt.Errorf("mapping Schema to SDK Field %q / Model %q: %+v", "Properties", "SolarSystemProperties", err)
+    }
+    
+    return nil
+}
+```
+
+As mentioned above, this implies the presence of an associated ModelToModel Mapping, which would then generate the mapping for the Model itself:
+
+```go
+func (r VirtualMachineResource) mapVirtualMachineResourceSchemaToVirtualMachineProperties(input VirtualMachineResourceSchema, output *virtualmachines.VirtualMachineProperties) error {
+    // ...
+    // DirectAssignment Field Mappings would be output here
+    // ...
+    
+    return nil
+}
+```
+
+### c. `Manual` Field Mappings
+
+> **Note:** At this point in time Manual Mappings are only partially configured and **need reworking** - as such this section explains the intention this type of Mapping, so that we can fix this in the future.
```

Comment:
```suggestion
> **Note:** At this point in time Manual Mappings are only partially configured and **need reworking** - as such this section explains the intention of this type of Mapping, so that we can fix this in the future.
```